x-pack/plugins/monitoring/public/components/logstash/pipeline_viewer/views/statement.js

```diff
@@ -6,11 +6,66 @@
 
 import React from 'react';
 import PropTypes from 'prop-types';
+import {
+  EuiButtonEmpty,
+  EuiCodeBlock,
+  EuiFlexItem } from '@elastic/eui';
 import { PluginStatement as PluginStatementModel } from '../models/pipeline/plugin_statement';
 import { CollapsibleStatement } from './collapsible_statement';
 import { IfElement } from '../models/list/if_element';
 import { PluginStatement } from './plugin_statement';
 
+function renderStatementName(name, onVertexSelected) {
+  return (
+    <EuiFlexItem
+      grow={false}
+      key="statementName"
+    >
+      <EuiButtonEmpty
+        color="text"
+        size="xs"
+        onClick={onVertexSelected}
+        flush="left"
+      >
+        <span className="pipelineViewer__conditional">{name}</span>
+      </EuiButtonEmpty>
+    </EuiFlexItem>
+  );
+}
+
+function renderIfStatement({ condition }, onVertexSelected) {
+  return [
+    renderStatementName('if', onVertexSelected),
+    (
+      <EuiFlexItem
+        key="ifContent"
+        grow={false}
+      >
+        <EuiCodeBlock
+          fontSize="s"
+          paddingSize="none"
+          transparentBackground={true}
+        >
+          {condition}
+        </EuiCodeBlock>
+      </EuiFlexItem>
+    )
+  ];
+}
+
+function getStatementBody({
+  isIf,
+  statement,
+  statement: { vertex },
+  onShowVertexDetails
+}) {
```

Comment:
Why does this function take one argument that's an object vs. multiple arguments corresponding to the values in the object?

Reply:
> I'm guessing I'm just missing something that's unique to this function or these particular parameters, maybe?

I think you make a good point actually. There's no reason for these keys to not just be their own params.

Reply:
@ycombinator: https://github.com/elastic/kibana/pull/20252/commits/a5dbb9880e568d47b02d725a2aefd43352741f17

I like this a lot better actually, it simplifies everything and removes some unnecessary abstraction. WDYT?